src/dirwatcher.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+const fsPromises = require('fs/promises');
+const { EventEmitter } = require('events');
+
+class DirWatcher extends EventEmitter {
+  constructor() {
+    super();
+    this.path = '';
+    this.directoryContents = new Map();
+    this.newDirectoryContents = new Map();
+    this.filesInDir = [];
+  }
+
+  static async readDir(path) {
+    return fsPromises.readdir(path);
+  }
+
+  static async readStats(path, fileNames) {
+    return fileNames.map(fileName => fsPromises.stat(`${path}/${fileName}`));
+  }
+
+  async saveState(path) {
+    // read dir file names
+    this.filesInDir = await DirWatcher.readDir(path);
+
+    // read stats
+    const stats = await DirWatcher.readStats(path, this.filesInDir);
+    const statsResolved = await Promise.all(stats);
+
+    // save stats to dir contents
+    return this.filesInDir.map((file, index) => (
+      [file, Number(statsResolved[index].mtime)]));
+  }
+
+  async watch(path, delay) {
+    this.path = path;
+    this.directoryContents = new Map([...await this.saveState(path)]);
+
+    // emit data change for first load if there is data:
+    if (this.directoryContents.size > 0) {
+      this.emit('data change', this.filesInDir);
+    }
+
+    // loop
+    const run = async () => {
+      this.newDirectoryContents = new Map([...await this.saveState(path)]);
+
+      // check for deleted files
+      const deletedFiles = [...this.directoryContents.keys()]
+        .filter(key => !this.newDirectoryContents.has(key));
+      if (deletedFiles.length > 0) {
+        this.emit('file deleted', deletedFiles);
```

Comment:
I think you should emit another kind of event for deleted files because right now your `Importer` will try to import even deleted files, which makes no sense. In such case it will be better to create some name constants like `FILE_CHANGED` and `FILE_DELETED`, export them as separate variables or static fields of `DirWatcher` class and use them instead of putting string literals. It will allow static checking and your IDE will throw error if you'll misspell name of such constant.